core/src/main/scala/pureconfig/DerivedConverters.scala

```diff
@@ -0,0 +1,268 @@
+  }
+
+  // traversable of types with an instance of ConfigConvert
+  implicit def deriveTraversable[T, F[T] <: TraversableOnce[T]](
+    implicit
+    configConvert: Lazy[ConfigConvert[T]],
+    cbf: CanBuildFrom[F[T], T, F[T]]) = new ConfigConvert[F[T]] {
+
+    override def from(config: ConfigValue): Either[ConfigReaderFailures, F[T]] = {
+      config match {
+        case co: ConfigList =>
+          val z: Either[ConfigReaderFailures, Builder[T, F[T]]] = Right(cbf())
+
+          // we called all the failures in the list
+          co.asScala.foldLeft(z) {
+            case (acc, value) =>
+              combineResults(acc, configConvert.value.from(value))(_ += _)
+          }.right.map(_.result())
+        case o: ConfigObject =>
+          val z: Either[ConfigReaderFailures, List[(Int, T)]] = Right(List.empty[(Int, T)])
+          def keyValueReader(key: String, value: ConfigValue): Either[ConfigReaderFailures, (Int, T)] = {
+            val keyResult = catchReadError(_.toInt)(implicitly)(key)(ConfigValueLocation(value)).left.flatMap(t => fail(CannotConvert(key, "Int",
+              s"To convert an object to a collection, it's keys must be read as Int but key $key has value" +
+                s"$value which cannot converted. Error: ${t.because}", ConfigValueLocation(value), Some(key))))
+            val valueResult = configConvert.value.from(value)
+            combineResults(keyResult, valueResult)(_ -> _)
+          }
+
+          o.asScala.foldLeft(z) {
+            case (acc, (str, v)) =>
+              combineResults(acc, keyValueReader(str, v))(_ :+ _)
+          }.right.map {
+            l =>
+              val r = cbf()
+              r ++= l.sortBy(_._1).map(_._2)
+              r.result()
+          }
+        case other =>
+          fail(WrongType(other.valueType().toString, "ConfigList or ConfigObject", ConfigValueLocation(other), None))
+      }
+    }
+
+    override def to(ts: F[T]): ConfigValue = {
+      ConfigValueFactory.fromIterable(ts.toList.map(configConvert.value.to).asJava)
+    }
+  }
+
+  implicit def deriveMap[T](implicit configConvert: Lazy[ConfigConvert[T]]) = new ConfigConvert[Map[String, T]] {
+
+    override def from(config: ConfigValue): Either[ConfigReaderFailures, Map[String, T]] = {
+      config match {
+        case co: ConfigObject =>
+          val z: Either[ConfigReaderFailures, Map[String, T]] = Right(Map.empty[String, T])
+
+          co.asScala.foldLeft(z) {
+            case (acc, (key, value)) =>
+              combineResults(acc, configConvert.value.from(value))((map, valueConverted) => map + (key -> valueConverted))
+          }
+
+        case other =>
+          fail(WrongType(other.valueType().toString, "ConfigObject", ConfigValueLocation(other), None))
+      }
+    }
+
+    override def to(keyVals: Map[String, T]): ConfigValue = {
+      ConfigValueFactory.fromMap(keyVals.mapValues(configConvert.value.to).asJava)
+    }
+  }
+
+  // used for products
```

Comment:
Nitpick: This comment doesn't feel useful. It doesn't tell me anything the name doesn't. It feels like either the comment ought to be expanded with some additional insight or deleted entirely.